### DISCUSSION TITLE
failed Scripting Simulations

### DISCUSSION THREAD

**Discussion by cyrus12sm** [simnibs/simnibs#204] [2. General]
Hello. I am new to Simnibs and I want to write a code using MATLAB or Python to simulate the electric field inside the head using TMS coils. My problem is that when I run the code in MATLAB I get the following error:
 {Error using run_simnibs (line 31)
there was an error running SimNIBS:
 usage: simnibs [-h] [--cpus CPUS] [--version] simnibs_file
simnibs: error: unrecognized arguments: Store\AppData\Local\Temp\tp52fca526_fd79_46a0_8c16_31dd03b0a3ca.mat

Error in tutorial_matlab (line 60)
run_simnibs(s)}

The codes that I run are the same codes that exist by default on the site and Simnibs file. I downloaded the latest version of Simnibs. My laptop's operating system is 64 bit Windows 10. Simnibs MATLAB function has been added in the MATLAB path.
I also have a problem in running the Python code and I encountered the following error:
{    import simnibs
ModuleNotFoundError: No module named 'simnibs'}
It seems that Python cannot find the installed modules. 
I have been trying to solve this problem for several weeks and have been unsuccessful. Please help me to solve this problem.
thank you.

**Comment by oulap** [comment]
Can you open the simnibs GUI and run a simulation there? You can see instructions on how to do that here:

https://simnibs.github.io/simnibs/build/html/tutorial/gui.html

This is just to check that the installation is fine, and the issue you're experiencing is strictly a scripting problem.

**Reply by cyrus12sm** [reply]
hello. yes I able to open GUI and run the simulation, there is no problem for that, but I have failed for any simulation by script.
I have checked the installation steps and the things mentioned on the site several times and there were no problems. 
At first, I was using MATLAB 2023, but to make sure that the problem is not with the MATLAB version, I installed the MATLAB 2019, but the error still persists.

**Reply by oulap** [reply]
If you run this in a terminal:

`simnibs_python -c "import simnibs; print('Hello')"`

does it print 'Hello' or throw an error?

**Reply by cyrus12sm** [reply]
I get this {
G:\sim\simnibs_env\lib\site-packages\requests\__init__.py:102: RequestsDependencyWarning: urllib3 (1.26.15) or chardet (5.2.0)/charset_normalizer (2.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({})/charset_normalizer ({}) doesn't match a supported "
Hello}

**Comment by oulap** [comment]
Seems like the simnibs_python installation works (apart from those warnings). 

Sometimes the postinstallation steps fail, can you check that the paths in these two files:

SIMNIBSDIR.m
SIMNIBSPYTHON.m

point to the correct locations in your system. 

You can find the files in: <simnibs_folder>/matlab_tools

**Reply by cyrus12sm** [reply]
function path=SIMNIBSDIR
% Function writen by SimNIBS postinstaller
path='C:\Users\Windows Store\AppData\Local\SimNIBS\simnibs_env\Lib\site-packages\simnibs';
end

function python_call=SIMNIBSPYTHON
% Function writen by SimNIBS postinstaller
python_call='"C:\Users\Windows Store\AppData\Local\SimNIBS\simnibs_env\pythonw.exe" -E -u ';
end

yes. the path of those two files already are correct.

**Reply by TorgeW** [reply]
```
function result = run_simnibs(session)

%Runs simnibs, given a session from sim_struct

% Guilherme Saturnino, 2018

%    This program is part of the SimNIBS package.
%    Please check on www.simnibs.org how to cite our work in publications.
%
%    Copyright (C) 2013-2018 Axel Thielscher, Andre Antunes, Guilherme Saturnino
%
%    This program is free software: you can redistribute it and/or modify
%    it under the terms of the GNU General Public License as published by
%    the Free Software Foundation, either version 3 of the License, or
%    any later version.
%
%    This program is distributed in the hope that it will be useful,
%    but WITHOUT ANY WARRANTY; without even the implied warranty of
%    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
%    GNU General Public License for more details.
%
%    You should have received a copy of the GNU General Public License
%    along with this program.  If not, see <http://www.gnu.org/licenses/>.

name_mat = [tempname,'.mat'];
save('-v7', name_mat, 'session');
[status,result] = system([simnibs_cli_call('run_simnibs') ' "' name_mat '"']);

if status ~= 0
    error('There was an error running SimNIBS:\n %s',result)
end
```

Can you please replace the contend of the "run_simnibs.m" file with what i posted and see if it works? :)

**Reply by cyrus12sm** [reply]
hello dear. I don't know how to express my gratitude for your help. I have been struggling to find a solution for several weeks, and no one else could identify the problem. I truly appreciate it. Thank you.
